### PR TITLE
Add page emoji support

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

Implement `icon` field on `page` entity. So it could be used to identify a page at first glance. 
It can be changed from sidebar, or page title.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202700524579012/f) _(internal)_
- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=1564%3A102859) _(internal)_

## 🐾 Next steps

- Support custom icons as page icons

## ❓ How to test this?

1. Checkout the branch / view the deployment
2. Open any page
3. Try using page icon picker 
4. Page icon should be updating on browser tab, page title and left sidebar

## 📹 Demo

https://user-images.githubusercontent.com/39553853/184602990-3e50d6f3-e150-4401-97e5-fdef6c93840e.mov

### CODE REVIEW

**Comment by nathggns** [comment]
Do we know how this interacts with https://github.com/hashintel/hash/pull/929 – do we need to add emoji picker support to the breadcrumbs later?

**Review by nathggns — APPROVED** [review]
This is really great work @yusufkinatas, well done! A couple of optional comments, but really good improvement!

File: packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/page-tree-item.tsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ const CustomContent = forwardRef((props: TreeItemContentProps, ref) => {
             color: [selected ? palette.gray[40] : "transparent"],
             "&:focus-visible, &:hover": {
               backgroundColor: palette.gray[selected ? 40 : 30],
-              color: palette.gray[selected ? 50 : 40],
+              color: `${palette.gray[selected ? 50 : 40]} !important`,
```
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/page-tree-item.tsx:147]: Yes, if the page is selected, the color was not applying before, and it caused a UI issue makes the icon invisible (same color with background). So `!important` solves that

File: packages/hash/frontend/src/components/EmojiPicker/EmojiPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import Picker from "@emoji-mart/react";
+import { Popover } from "@hashintel/hash-design-system";
+import { BaseEmoji, PickerProps } from "emoji-mart";
+import { bindPopover, PopupState } from "material-ui-popup-state/core";
+import { FunctionComponent } from "react";
+
+const TypedPicker = Picker as FunctionComponent<PickerProps>;
```
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply packages/hash/frontend/src/components/EmojiPicker/EmojiPicker.tsx:7]: Yeah, definitely makes sense, thanks for the suggestion!

File: packages/hash/shared/src/queries/page.queries.ts [hunk 3]
```diff
@@ -30,6 +31,12 @@ export const updatePage = gql`
     ) {
       accountId
       entityId
+      properties {
+        title
+        icon
+        archived
+        pageEntityId
```
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply packages/hash/shared/src/queries/page.queries.ts]: Makes sense, applied at https://github.com/hashintel/hash/pull/933/commits/786763c9ce5080498b061f654ba528760eb6c43e

File: patches/@types+emoji-mart+3.0.9.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+diff --git a/node_modules/@types/emoji-mart/dist-es/utils/shared-props.d.ts b/node_modules/@types/emoji-mart/dist-es/utils/shared-props.d.ts
+index ba8f035..f1062df 100755
+--- a/node_modules/@types/emoji-mart/dist-es/utils/shared-props.d.ts
++++ b/node_modules/@types/emoji-mart/dist-es/utils/shared-props.d.ts
+@@ -63,6 +63,7 @@ export interface PickerProps {
+     /** NOTE: default is not preventable */
+     onClick?(emoji: EmojiData, e: React.MouseEvent<HTMLElement>): void;
+     onSelect?(emoji: EmojiData): void;
++    onEmojiSelect?(emoji: EmojiData): void;
```
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply patches/@types+emoji-mart+3.0.9.patch:9]: Added a comment line that explains the reason behind this typecast here https://github.com/hashintel/hash/pull/933/commits/b65658d85f1d40782694747b44019bc5a529ccc2

File: packages/hash/frontend/src/components/EmojiPicker/EmojiPicker.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (yusufkinatas) [reply packages/hash/frontend/src/components/EmojiPicker/EmojiPicker.tsx]: Applied the suggestion here https://github.com/hashintel/hash/pull/933/commits/066a53fe83dee5fcc18e113b7a9f7c24de3ae234

**Comment by yusufkinatas** [comment]
> Do we know how this interacts with #929 – do we need to add emoji picker support to the breadcrumbs later?

Yes, we need to render the `EmojiPicker` inside the breadcrumbs as well. Should we create a follow-up ticket for this?

**Review by nathggns — APPROVED** [review]
Great work @yusufkinatas

**Comment by nathggns** [comment]
@yusufkinatas yes please create a follow up

**Pull request merged**